sdk/storage/azure-storage-blobs/src/blob_lease_client.cpp

```diff
@@ -52,12 +53,16 @@ namespace Azure { namespace Storage { namespace Blobs {
       protocolLayerOptions.IfModifiedSince = options.AccessConditions.IfModifiedSince;
       protocolLayerOptions.IfUnmodifiedSince = options.AccessConditions.IfUnmodifiedSince;
 
-      if (options.AccessConditions.IfMatch.HasValue()
-          || options.AccessConditions.IfNoneMatch.HasValue()
-          || options.AccessConditions.TagConditions.HasValue())
-      {
-        std::abort();
-      }
+      AZURE_ASSERT_MSG(
+          !options.AccessConditions.IfMatch.HasValue(),
+          "Blob container lease doesn't support If-Match condition");
+      AZURE_ASSERT_MSG(
+          !options.AccessConditions.IfNoneMatch.HasValue(),
+          "Blob container lease doesn't support If-None-Match condition");
+      AZURE_ASSERT_MSG(
+          !options.AccessConditions.TagConditions.HasValue(),
+          "Blob container lease doesn't support tag condition");
```

Comment:
Don't use ASSERT_MSG, use ASSERT alone.  ASSERT_MSG is only for when the condition is well known to be false, like the non_implemented macro

Reply:
But `AZURE_ASSERT` doesn't print a message, which can be helpful to customers if they hit an assertion failure.

Reply:
ASSUTE_ASSERT will print a message which will include the condition, file, line number, etc.  So, people will see this in that message:  `!options.AccessConditions.TagConditions.HasValue()` which is good enough to understand that, as a contract, tagConditions can't be there when calling that method.  The message would be mostly ignored or redundant.

The message is relevant when the assert is just something like `false`,  in that case, the message we get would contain only the word `false`.  That's the case where the message would help and useful.

Also, we want to call AZURE_ASSERT_MSG with `false` directly so we are sure that the message will be used.  If we pass a condition, and the condition is true, the message is just ignored and never used, but we are still building it as part of the Debug build.  

On release build, no matter if you pass a message or not, the ASSERT is ignored to just call abort() directly. So we won't see any message or anything on Production.